build/scripts/build.ps1

```diff
@@ -17,73 +17,88 @@
-    Write-Host "  -build                    Build Roslyn.sln"
-    Write-Host "  -official                 Perform an official build"
-    Write-Host "  -bootstrap                Build using a bootstrap Roslyn"
-    Write-Host "  -sign                     Sign our binaries"
+    Write-Host "  -build                    Build main solution"
+    Write-Host "  -rebuild                  Rebuild main solution"
     Write-Host "  -pack                     Build NuGet packages, VS insertion manifests and installer"
-    Write-Host "  -deployExtensions         Deploy built vsixes"
-    Write-Host "  -binaryLog                Create binary log for every MSBuild invocation"
-    Write-Host "  -procdump                 Monitor test runs with procdump"
-    Write-Host "  -skipAnalyzers            Do not run analyzers during build operations"
-    Write-Host "  -checkLoc                 Check that all resources are localized"
+    Write-Host "  -sign                     Sign our binaries"
+    Write-Host "  -publish                  Publish build artifacts (e.g. symbols)"
+    Write-Host "  -launch                   Launch Visual Studio in developer hive"
+    Write-Host "  -help                     Print help and exit"
     Write-Host ""
-    Write-Host "Test options"
+    Write-Host "Test actions"
     Write-Host "  -test32                   Run unit tests in the 32-bit runner"
     Write-Host "  -test64                   Run units tests in the 64-bit runner"
-    Write-Host "  -testDesktop              Run desktop unit tests"
+    Write-Host "  -testDesktop              Run Desktop unit tests (short: -test)"
     Write-Host "  -testCoreClr              Run CoreClr unit tests"
     Write-Host "  -testVsi                  Run all integration tests"
-    Write-Host "  -testVsiNetCore           Run just dotnet core integration tests"
     Write-Host "  -testIOperation           Run extra checks to validate IOperations"
     Write-Host ""
-    Write-Host "Special Test options"
-    Write-Host "  -testDeterminism          Run determinism tests"
+    Write-Host "Advanced settings:"
+    Write-Host "  -ci                       Set when running on CI server"
+    Write-Host "  -official                 Set when building an official build"
+    Write-Host "  -bootstrap                Build using a bootstrap compilers"
+    Write-Host "  -msbuildEngine <value>    Msbuild engine to use to run build ('dotnet', 'vs', or unspecified)."
+    Write-Host "  -procdump                 Monitor test runs with procdump"
+    Write-Host "  -skipAnalyzers            Do not run analyzers during build operations"
+    Write-Host "  -prepareMachine           Prepare machine for CI run, clean up processes after build"
+    Write-Host ""
+    Write-Host "Command line arguments starting with '/p:' are passed through to MSBuild."
+}
+
+if ($help -or (($properties -ne $null) -and ($properties.Contains("/help") -or $properties.Contains("/?")))) {
```

Comment:
Why put this here vs. the main `try / finally` body?

Reply:
Will move.

Reply:
Moved to Process-Arguments